core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -913,6 +940,18 @@ object IO extends Serializable {
       )
     )
 
+  /**
+   * The moral equivalent of `if (p) exp`
+   */
+  final def when[E](b: Boolean, io: IO[E, Unit]): IO[E, Unit] =
+    if (b) io else IO.unit
+
+  /**
+   * The moral equivalent of `if (p) exp` when `p` has side-effects
+   */
+  final def whenM[E](b: IO[Nothing, Boolean], io: IO[E, Unit]): IO[E, Unit] =
```

Comment:
> final def when[E](b: Boolean, io: IO[E, Unit])

Suggestion: maybe this could have split argument lists for convenience?

`final def when[E](b: Boolean)(io: IO[E, Unit])`

Possibly with a partially applied class if inference is bad without it.